qcow.c

```diff
@@ -307,15 +307,17 @@ static int qcow2_probe(struct bdev *bdev, int dirfd, const char *pathname)
 	tcmu_dbg("%s\n", __func__);
 
 	if (faccessat(dirfd, pathname, R_OK|W_OK, AT_EACCESS) == -1) {
-		perror("no file access");
+		tcmu_err("faccessat failed dirfd %d, pathname %s, errno %d\n",
+			 dirfd, pathname, errno);
 		return -1;
 	}
 	if ((fd = openat(dirfd, pathname, O_RDONLY)) == -1) {
-		perror("openat failed");
+		tcmu_err("openat failed dirfd %d, pathname %s, errno %d\n",
+			 dirfd, pathname, errno);
 		return -1;
 	}
 	if (pread(fd, &head, sizeof(head), 0) == -1) {
-		perror("read fail");
+		tcmu_err("read failed fd: %d\n", fd);
```

Comment:
It would be more helpful if you printed the pathname and errno.  I'm not sure that the fd is overly useful here.